test/compilable/test17502.d

```diff
@@ -0,0 +1,30 @@
+class Foo
+{
+    auto foo()
+    out {}
+    body {}
+
+    auto bar()
+    out { assert (__result > 5); }
```

Comment:
Well, it's an internal compiler test, I guess, so I was going with the thoughts that whoever removes `__result`, should do it contently and remove this test. Reason I've added it here in the first place is that this patch touches the code responsible for the generation of this identifier.